Shop_Unity_base/Assets/Scripts/Shop/Model/Items/IClonable.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Model.Items
+{
+    public interface IClonable
+    {
+        IClonable Clone();
+    }
```

Comment:
```suggestion
    public interface IClonable { }
    public interface IClonable<T> : IClonable where T : IClonable 
    {
        T Clone();
    }
```

Reply:
Prevent explicit casting. It's a bit circular but it works.

Reply:
For example, use like so:
```csharp
    public class Weapon : Item, IClonable<Weapon> { }
```

Reply:
Hmmmm alright